sandbox.go

```diff
@@ -676,23 +676,68 @@ func (sb *sandbox) SetKey(basePath string) error {
 
 func (sb *sandbox) EnableService() error {
 	logrus.Debugf("EnableService %s START", sb.containerID)
+
 	for _, ep := range sb.getConnectedEndpoints() {
-		if ep.enableService(true) {
+		if !ep.isServiceEnabled() {
 			if err := ep.addServiceInfoToCluster(sb); err != nil {
-				ep.enableService(false)
 				return fmt.Errorf("could not update state for endpoint %s into cluster: %v", ep.Name(), err)
 			}
+			// enable service on the endpoint copy in the sandbox
+			ep.enableService(true)
+
+			n, err := ep.getNetworkFromStore()
+			if err != nil {
+				logrus.Warnf("could not enable service on sandbox:%s,endpoint:%s,err: %v", sb.ID(), ep.Name(), err)
+				continue
+			}
+
+			store := n.getController().getStore(ep.DataScope())
+
+			if store == nil {
+				logrus.Warnf("store not found for scope %s on enable service on endpoint:%s", ep.DataScope(), ep.Name())
+			}
+
+			//Write the ep copy to the store
+			ep.enableService(true)
+			if err := n.getController().updateToStore(ep); err != nil {
+				break
```

Comment:
also when we break here basically the service got already updated but now the flag will remain at false, so there would not be any disable service happening neither in the case of sbLeave